payment.php

```diff
@@ -125,23 +125,27 @@ function generate_address($blockonomics, $ca) {
 		 * Status 0 or 1: Order has a pending payment, show info about pending transaction and disallow payment
 		 * Status 2: Order has received an underpayment, generate new address
 		 */
+		$confirmations = $blockonomics->getConfirmations();
 		if($existing_order['status'] == -1) {
 			$btc_address = $existing_order['address'];
-		} elseif ($existing_order['status'] == 0 || $existing_order['status'] == 1) {
+			$btc_amount = $blockonomics->getBitcoinAmount($fiat_amount, $currency);
+			$ca->assign('btc_amount', $btc_amount / 1.0e8);
+			$blockonomics->updateOrderExpected($order_id, $btc_amount, $fiat_amount);
+		} elseif ($existing_order['status'] < $confirmations) {
 			$ca->assign('pending', true);
 			$ca->assign('txid', $existing_order['txid']);
-		} elseif ($existing_order['status'] == 2) {
+		} elseif ($existing_order['status'] >= $confirmations) {
 			$btc_address = generate_address($blockonomics, $ca);
 			$blockonomics->updateOrderAddress($order_id, $btc_address);
+			$btc_amount = $blockonomics->getBitcoinAmount($fiat_amount, $currency);
+			$ca->assign('btc_amount', $btc_amount / 1.0e8);
+			$blockonomics->updateOrderExpected($order_id, $btc_amount, $fiat_amount);
```

Comment:
Is it not necessary to existing order status to *-1* again?

Reply:
It is done inside the function https://github.com/blockonomics/whmcs-bitcoin-plugin/blob/324c3433c7bdce74da6539e48996ca5f4894b6e9/modules/gateways/Blockonomics/Blockonomics.php#L462